### DISCUSSION TITLE
Drawing on map

### DISCUSSION THREAD

**Discussion by alexandrdmitrievich** [openlayers/openlayers#16144] [Q&A]
Hi there! I was wondering if it's possible to draw and erase lines on a map, like you can on a canvas? I know it's probably possible, but I'm just not sure how to use a regular canvas element as a layer on the map and draw raster, not vector lines. Any advice would be greatly appreciated!

**Comment by mprins** [comment]
https://openlayers.org/en/latest/examples/draw-shapes.html and https://openlayers.org/en/latest/examples/draw-freehand.html are some examples, but there are many more: https://openlayers.org/en/latest/examples/?q=draw

**Reply by alexandrdmitrievich** [reply]
It's vector drawings, each shape is basically a feature on layer. I am seaking for raster drawings.
There is no easy way to erase middle of line, or particially change opacity of a line, or any indepth freehand doodling. 
I am looking for analog to drawing on StaticImage layer. I wrote it, but it hella slow, because I have to recreate source on each event like 
```
.setSource(new Static({url: canvas.toDataURL(), projection: pj, imageExtent: ex}))
``` 

Probably I should try to draw on canvas as overlay above map. But I dont sure that overlay will be 'glued' to map coordinates.
